clang/test/Driver/sycl-offload-tempfile.cpp

```diff
@@ -0,0 +1,64 @@
+// REQUIRES: x86-registered-target
+// UNSUPPORTED: system-windows
+// 
+// Test to determine that temp file cleanup is working for fat static archives
+// creates a single file fat archive then uses that
+// RUN: rm -rf %t
+// RUN: mkdir %t
+// Build static library
+// RUN: %clang -DBUILD_B -fsycl -c -o %t_obj.o %s
+// RUN: ar cr %t_lib.a %t_obj.o
+// Build main object
+// RUN: %clang -DBUILD_A -fsycl -c -o %t_main.o %s
+// Build final binary, overriding output temp location
+// RUN: env TMPDIR=%t TEMP=%t TMP=%t                                     \
+// RUN: not %clang -fsycl %t_main.o -foffload-static-lib=%t_lib.a
+// RUN: not ls %t/*
+#ifdef BUILD_A
+#include <CL/sycl.hpp>
```

Comment:
Adding dependency to SYCL project in clang tests is a bad idea.
Clang tests must be self-contained.

I suggest simplifying the test to avoid any includes if possible.

If not, you can use stub headers as CodeGenSYCL tests: https://github.com/intel/llvm/blob/sycl/clang/test/CodeGenSYCL/kernel_functor.cpp uses https://github.com/intel/llvm/blob/sycl/clang/test/CodeGenSYCL/Inputs/sycl.hpp.

Reply:
Will do - Just need to test removal of files from unbundler of a fat static library, which doesn't need anything from SYCL.